CHANGELOG.md

```diff
@@ -0,0 +1,50 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
+and follows a [Backwards Compatability Policy]
+(https://docs.solana.com/developing/backwards-compatibility)
+
+## Unreleased
```

Comment:
The "Unreleased" and "Released" section headers are still somewhat problematic for tagging, as they'll require a commit BEFORE tagging. 

I was thinking we'd just have a single list, with 1.17.0 at the top in this case, where new entries are added. `./scripts/increment-cargo-version.sh` then adds a new entry above it, be it 1.17.1 or 1.18.0, as it runs to update the rest of the version numbers post-tagging

Reply:
it'd be nice if we can denote the "working" release somehow. as described, that information would need recovered from another source. i'm struggling to see how we do that without another commit (and obnoxious additional release coordination) though

Reply:
I'll have a go at a script to generate the releases. My preference is to avoid BuildKite but open to that if anyone sees benefit of doing it there.

Reply:
Cool. A buildkite job with an [input step](https://buildkite.com/docs/pipelines/input-step) to specify the kind of release seems like it could be pretty convenient: no need to check out the exact tree to ship locally, or dork around with a github access token

Reply:
we decided (last week) that this script would be the way forward, but ought not block this pr.  resolving this convo accordingly